Consistent casing and smart contract facelift

czepluch opened the pull request:
This PR makes sure that all variable names use consistent casing. @LefterisJP and I decided to use snake case for variable names and let the function names me mixed case, since this is a nice way to distinguish smart contract calls in the python code.

I also made a few changes to make the smart contracts look a bit better. This was mainly done by splitting up some very long function declarations.

LefterisJP commented on raiden/smart_contracts/ChannelManagerLibrary.sol:
```diff
@@ -21,19 +21,25 @@ library ChannelManagerLibrary {
+    /// @param node_address (address) the address of the node
+    /// @return (address[]) of the channel's addresses that node_address participates.
+    function getChannelsForNode(Data storage self, address node_address)
+        constant
```
In this and in other places where the modifier and function attributes go to a different line I don't think it's needed. Only if it's too big should you do that. And then probably not each modifier on its own line, could be an overkill ... but I am not sure. I can't say I dislike everything on its own line.

czepluch replied:
So you'd prefer `constant returns (address[])` on one line here instead of splitting it up? Just to be sure that I understand what you mean.

LefterisJP commented on raiden/smart_contracts/ChannelManagerLibrary.sol:
```diff
@@ -50,30 +56,36 @@ library ChannelManagerLibrary {
+        address partner,
+        uint settle_timeout
+    )
+        returns (address)
```
The `returns` should be right after the parentheses, it can be mistaken for the body of the function

czepluch replied:
I don't agree. I think it's very clear that it's not the body of the function due to the curly bracket in the next line.

LefterisJP replied:
no I strongly disagree here.

Reasoning:

When one skims through the code quickly (which is how you should read code on your first pass) the eye searches for indented blocks in order to identify function bodies and quickly deduce functionality. 

Here there is a change of indentation from the closing parentheses to the `returns`.

Perhaps if you don't want it in the same line, a compromise would be to close the parentheses after `uint settle_timeout` so that at least they are all on the same level?

LefterisJP replied:
Can you put the parentheses nex to `settle_timeout` here?

LefterisJP commented on raiden/smart_contracts/NettingChannelLibrary.sol:
```diff
@@ -45,53 +45,60 @@ library NettingChannelLibrary {
+        address channel_address,
+        uint256 amount
+    )
+        returns (bool success, uint256 balance)
```
same here, returns right next to the parentheses to show it's part of the declaration and not the body.

LefterisJP replied:
As I said please put the parentheses next to the last argument.

LefterisJP commented on raiden/smart_contracts/NettingChannelLibrary.sol:
```diff
@@ -112,36 +119,44 @@ library NettingChannelLibrary {
     }
 
-    function addressAndBalance(Data storage self) constant returns (address participant1, uint balance1, address participant2, uint balance2) {
+    function addressAndBalance(Data storage self)
```
Same thing here. I won't repeat the same comment in the subsequent locations where the same thing happens but what do you think about this:

```
function addressAndBalance(Data storage self) constant returns (
    address participant1,
    uint balance1,
    address participant2,
    uint balance2) {

    .... 
    bla bla 
    .... 
    function body
    ... 
} 
```

czepluch replied:
I can live with that, I just followed the suggested guidelines from the solidity documentation like this:
```
function thisFunctionNameIsReallyLong(address x, address y, address z)
    public
    onlyowner
    priced
    returns (address)
{
    doSomething();
}

function thisFunctionNameIsReallyLong(
    address x,
    address y,
    address z,
)
    public
    onlyowner
    priced
    returns (address)
{
    doSomething();
}
```
But I am open to making some changes that makes it a bit more compact like you suggest.

czepluch commented:
@LefterisJP so except for the fact that I fu**ed the rebasing up here too, this should be good to merge now. Fixed the suggestions you made and fixed all the naming in tests that aren't smart contract tests, so all tests are passing again now.

LefterisJP commented:
@czepluch Great! Can you fix the rebasing though? It's quite simple as I told you in chat.

czepluch commented:
@LefterisJP Rebasing fixed. Please review.

LefterisJP commented on raiden/smart_contracts/NettingChannelLibrary.sol:
```diff
@@ -157,31 +172,36 @@ library NettingChannelLibrary {
+        address caller_address,
+        bytes first_encoded,
+        bytes second_encoded
+    ) {
```
Same here, parentheses next to the last argument.

czepluch replied:
@LefterisJP I think these changes got lost in my rebase chaos earlier, because I remember doing them. Nonetheless, I've pushed them again now and learned an good git lesson :)

LefterisJP commented on raiden/smart_contracts/NettingChannelLibrary.sol:
```diff
@@ -112,36 +119,44 @@ library NettingChannelLibrary {
+        address participant1,
+        uint balance1,
+        address participant2,
+        uint balance2
```
Same here, parentheses next to the last argument.
